scheduler/feasible.go

```diff
@@ -216,6 +216,27 @@ func (h *HostVolumeChecker) hasVolumes(n *structs.Node) bool {
 			if !capOk {
 				return false
 			}
+			if req.Sticky {
+				// NOTE: surely there is a better way to find the right alloc?
+				// Should we perhaps search for allocs by job? Could there be a
+				// situation in which there are non-terminal allocations
+				// belonging to the job in question that are sticky, have the
+				// volume IDs that match what the node offers and should not
+				// end up in this check?
+				allocs, err := h.ctx.ProposedAllocs(n.ID)
```

Comment:
The only other place we use `ctx.ProposedAllocs` for feasibility checking is in the distinct hosts checker, because in that case we need to know about _all_ the allocations on the node, including those that might already exist. Do we really want to look at the old allocs here?